src/com/palmergames/bukkit/towny/utils/JailUtil.java

```diff
@@ -82,11 +86,17 @@ public static void jailResident(Resident resident, Jail jail, int cell, int hour
 		resident.setJail(jail);
 		resident.setJailCell(cell);
 		resident.setJailHours(hours);
+		resident.setJailBail(bail);
 		resident.save();
 		TownyUniverse.getInstance().getJailedResidentMap().add(resident);
 		
-		// Tell the resident how long they've been jailed for.
-		TownyMessaging.sendMsg(resident, Translatable.of("msg_you've_been_jailed_for_x_hours", hours));
+		// Tell the resident how long they've been jailed for and provide bail information if allowing bail and using economy
+		if (TownySettings.isAllowingBail() && TownyEconomyHandler.isActive())
+		{
+			TownyMessaging.sendMsg(resident, Translatable.of("msg_you've_been_jailed_for_x_hours", hours, Translatable.of("msg_you_have_been_jailed_your_bail_is_x", bail)));
+		} else {
+			TownyMessaging.sendMsg(resident, Translatable.of("msg_you've_been_jailed_for_x_hours", hours));
+		}

```

Comment:
```suggestion
		// Tell the resident how long they've been jailed for and provide bail information if allowing bail and using economy
		TownyMessaging.sendMsg(resident, Translatable.of("msg_you've_been_jailed_for_x_hours", hours, 
		if (TownySettings.isAllowingBail() && bail > 0 && TownyEconomyHandler.isActive())
			Translatable.of("msg_you_have_been_jailed_your_bail_is_x", bail)));

```

Reply:
Don't need to use an else when the message should always be shown. Also made it only show if the bail is greater than 0.

Reply:
Suggested change actioned on in latest commit.